### PR TITLE
alter streaming jobs with dependencies

### PR DESCRIPTION
**Pull request by fuyufjh**
## Info

- **Description**

Add an example of altering a streaming job with dependencies.

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves [ Provide a link to the relevant doc issue here, if applicable. ]

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/manage/alter-streaming.md [hunk 1]
```diff
@@ -27,46 +27,102 @@ The new column will be `NULL` for existing records.
...
-If your sink is created based on a materialized view using the CREATE SINK ... FROM ... statement, you have the option to specify `without_backfill = true` to exclude existing data.
+If your sink is created based on a materialized view using the `CREATE SINK ... FROM ...` statement, you have the option to specify `without_backfill = true` to exclude existing data.
 
 ```sql
 CREATE SINK ... FROM ... WITH (without_backfill = true).
 ```
 
+## Alter streaming jobs with dependencies
+
+If the modified materialized view is depended upon by other materialized views or sinks, or if multiple materialized views or sinks with dependencies need to be modified simultaneously, similarly, first create all the new ones and then drop all the old ones.
```
Comment (neverchanje) [inline docs/manage/alter-streaming.md]: This paragraph reads not very well. @hengm3467 Please help refine it.

I think Eric intends to illustrate that altering stream jobs with dependencies requires a series of cascading altering operations. 

In a case where there are 3 mviews, A -> B -> C, modifying A requires modifying B and C cascadingly. 

BTW, I also wanted to add a new topic regarding how to leverage dbt to simplify the cascade altering. I will document this in a new PR

File: docs/manage/alter-streaming.md [hunk 2]
```diff
@@ -27,46 +27,102 @@ The new column will be `NULL` for existing records.
...
+
+Let's continue with the previous example, but suppose the upstream `orders` is not a table but another materialized view, derived from tables `order_items` and `price`.
```
Comment (neverchanje) [inline docs/manage/alter-streaming.md:74]: Probably we need to add a visual for this case.

Comment (fuyufjh) [reply docs/manage/alter-streaming.md:74]: Make sense. Let me draw one.

File: docs/manage/alter-streaming.md (hunk 1 above)
Comment (hengm3467) [reply docs/manage/alter-streaming.md]: @WanYixian Please help reword this paragraph. You can see the example below for reference. If A is the level-0 MV, and B is level-1 MV, meaning B is built on top of A, and C is level-2 MV. To modify A, we need to create A_new, then B_new based on A_new, and C_new based on B_new, and then drop A, B, and C, and then rename A_new to A, B_new to B, and C_new to C.

File: docs/manage/alter-streaming.md [hunk 3]
```diff
@@ -27,46 +27,106 @@ The new column will be `NULL` for existing records.
...
-If your sink is created based on a materialized view using the CREATE SINK ... FROM ... statement, you have the option to specify `without_backfill = true` to exclude existing data.
+If your sink is created based on a materialized view using the `CREATE SINK ... FROM ...` statement, you have the option to specify `without_backfill = true` to exclude existing data.
 
 ```sql
 CREATE SINK ... FROM ... WITH (without_backfill = true).
 ```
 
+## Alter streaming jobs with dependencies
+
+If the modified materialized view is depended upon by other materialized views or sinks, or if multiple materialized views or sinks with dependencies need to be modified simultaneously, similarly, first create all the new ones and then drop all the old ones.
```
Comment (WanYixian) [inline docs/manage/alter-streaming.md]:
```suggestion
If other materialized views or sinks depend upon the modified materialized view, or if multiple materialized views or sinks with dependencies need updating simultaneously, similarly, create all the new ones and then drop all the old ones.
```

**Review by WanYixian — APPROVED** [review]

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**